cdap-common/src/main/java/io/cdap/cdap/common/internal/remote/RemoteClientFactory.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.common.internal.remote;
+
+import io.cdap.cdap.security.spi.authentication.AuthenticationContext;
+import io.cdap.common.http.HttpRequestConfig;
+import org.apache.twill.discovery.DiscoveryServiceClient;
+
+import javax.inject.Inject;
+
+/**
+ * A factory to create {@link RemoteClient}.
+ */
+public class RemoteClientFactory {
+
+  private final DiscoveryServiceClient discoveryClient;
+  private final AuthenticationContext authenticationContext;
+
+  @Inject
+  public RemoteClientFactory(DiscoveryServiceClient discoveryClient, AuthenticationContext authenticationContext) {
```

Comment:
If it only get new in test, annotate it with `@VisibleForTesting`

Reply:
Done